doc-files/release-notes-21.md

```diff
@@ -0,0 +1,153 @@
+```
+
+This change allows applications to manage event handlers for all `EventTarget` implementations.
+
+Most applications will not run into any backward compatibility problems with this change. The change is fully binary compatible and, for most use cases, it is also source compatible.
+
+A compilation error will occur in one of the following two unlikely cases:
+
+* An application class extends `Menu`, `MenuItem`, `TableColumnBase`, or `TreeItem`, and overrides `addEventHandler` or `removeEventHandler`
+* An application class that is not a subclass of `Node` implements their own `addEventHandler` or `removeEventHandler` using a signature other than that of the newly added methods (for example, copying the same incorrect pattern that `MenuItem` used prior to this release)
+
+In these cases, the declared methods must be changed to conform to the updated interface method signature.
+
+See [JDK-8306021](https://bugs.openjdk.org/browse/JDK-8306021) for more information.
+
+## List of Enhancements
+
+Issue key|Summary|Subcomponent
+---------|-------|------------
+[JDK-8284542](https://bugs.openjdk.org/browse/JDK-8284542)|[Win] Missing attribute for toggle state of CheckBox in CheckBoxTreeItem|accessibility
+[JDK-8301312](https://bugs.openjdk.org/browse/JDK-8301312)|Create implementation of NSAccessibilityButton protocol|accessibility
+[JDK-8283063](https://bugs.openjdk.org/browse/JDK-8283063)|Optimize Observable{List/Set/Map}Wrapper.retainAll/removeAll|base
+[JDK-8304439](https://bugs.openjdk.org/browse/JDK-8304439)|Subscription based listeners|base
+[JDK-8304960](https://bugs.openjdk.org/browse/JDK-8304960)|ObservableListBase should defer constructing ListChangeBuilder|base
+[JDK-8305885](https://bugs.openjdk.org/browse/JDK-8305885)|Use ReadOnly*PropertyBase class where possible|base
+[JDK-8306021](https://bugs.openjdk.org/browse/JDK-8306021)|Add event handler management to EventTarget|base
+[JDK-8090647](https://bugs.openjdk.org/browse/JDK-8090647)|Mnemonics: on windows we should cancel the underscore latch when an app loses focus|controls
+[JDK-8091153](https://bugs.openjdk.org/browse/JDK-8091153)|Customize the Table Button Menu|controls
+[JDK-8091419](https://bugs.openjdk.org/browse/JDK-8091419)|TableView: invoke table menu button programmatically|controls
+[JDK-8307960](https://bugs.openjdk.org/browse/JDK-8307960)|Create Table Column PopupMenu lazily|controls
+[JDK-8309470](https://bugs.openjdk.org/browse/JDK-8309470)|Potential performance improvements in VirtualFlow|controls
+[JDK-8290765](https://bugs.openjdk.org/browse/JDK-8290765)|Remove parent disabled/treeVisible listeners|graphics
+[JDK-8307363](https://bugs.openjdk.org/browse/JDK-8307363)|TextFlow.underlineShape()|graphics
+[JDK-8299756](https://bugs.openjdk.org/browse/JDK-8299756)|Minor updates in CSS Reference|other
+[JDK-8306648](https://bugs.openjdk.org/browse/JDK-8306648)|Update the JavaDocs to show the NEW section and DEPRECATED versions|other
```

Comment:
Should we mention somewhere that this doesn't always work properly because of [JDK-8309765](https://bugs.openjdk.org/browse/JDK-8309765)?

Reply:
This bug doesn't rise to the level of "Important Changes", which are mainly intended to alert developers to possible compatibility issues, so there isn't a good place to add it. I suppose we could add a "Known Bugs" section to list bugs that are either regressions from a prior release or are bugs in newly added features, but I don't want to do that this late in the cycle, especially not without discussing the general idea of whether or not to list known bugs.